source/common/http/http2/codec_impl.cc

```diff
@@ -37,6 +37,17 @@ bool Utility::reconstituteCrumbledCookies(const HeaderString& key, const HeaderS
   return true;
 }
 
+void initializeNghttp2Logging() {
+  nghttp2_set_debug_vprintf_callback([](const char* format, va_list args) {
+    char buf[2048];
+    const int n = ::vsnprintf(buf, sizeof(buf), format, args);
+    if (n >= 1 && static_cast<size_t>(n) < sizeof(buf) && buf[n - 1] == '\n') {
```

Comment:
Can you add a quick comment here? I guess nghttp2 wants a newline, but then we get double new lines?